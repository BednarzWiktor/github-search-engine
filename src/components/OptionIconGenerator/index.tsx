import React from 'react';

import { Storage, Person } from '@material-ui/icons';

import { OptionIconGeneratorProps as Props } from '../../types/propTypes';

const OptionIconGenerator = ({ origin }: Props) => {
    switch (origin) {
        case 'repo': return <Storage />;
        case 'user': return <Person />;
        default: return null;
    };
};

export default OptionIconGenerator;