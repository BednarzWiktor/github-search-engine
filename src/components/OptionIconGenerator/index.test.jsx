import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { mount, configure } from 'enzyme';

import toJson from 'enzyme-to-json';

import OptionIconGenerator from './';

configure({adapter: new Adapter()});
describe('OptionIconGenerator', () => {
    const userWrapper = mount(<OptionIconGenerator origin="user"/>);
    const repoWrapper = mount(<OptionIconGenerator origin="repo"/>);

    it('renders without crashing', () => {
        expect(toJson(userWrapper)).toMatchSnapshot();
        expect(toJson(repoWrapper)).toMatchSnapshot();
    });

    it('renders appropriate icon based on passed in origin prop', () => {
        const user = userWrapper.find('path').at(0);
        const repo = repoWrapper.find('path').at(0);

        expect(user.html()).toEqual(
            `<path d=\"M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z\"></path>`
        );
        expect(repo.html()).toEqual(
            `<path d=\"M2 20h20v-4H2v4zm2-3h2v2H4v-2zM2 4v4h20V4H2zm4 3H4V5h2v2zm-4 7h20v-4H2v4zm2-3h2v2H4v-2z\"></path>`
        );
    });
});