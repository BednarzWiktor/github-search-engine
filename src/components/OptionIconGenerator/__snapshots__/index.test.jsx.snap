// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionIconGenerator renders without crashing 1`] = `
<OptionIconGenerator
  origin="user"
>
  <ForwardRef>
    <WithStyles(ForwardRef(SvgIcon))>
      <ForwardRef(SvgIcon)
        classes={
          Object {
            "colorAction": "MuiSvgIcon-colorAction",
            "colorDisabled": "MuiSvgIcon-colorDisabled",
            "colorError": "MuiSvgIcon-colorError",
            "colorPrimary": "MuiSvgIcon-colorPrimary",
            "colorSecondary": "MuiSvgIcon-colorSecondary",
            "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
            "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
            "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
            "root": "MuiSvgIcon-root",
          }
        }
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </ForwardRef(SvgIcon)>
    </WithStyles(ForwardRef(SvgIcon))>
  </ForwardRef>
</OptionIconGenerator>
`;

exports[`OptionIconGenerator renders without crashing 2`] = `
<OptionIconGenerator
  origin="repo"
>
  <ForwardRef>
    <WithStyles(ForwardRef(SvgIcon))>
      <ForwardRef(SvgIcon)
        classes={
          Object {
            "colorAction": "MuiSvgIcon-colorAction",
            "colorDisabled": "MuiSvgIcon-colorDisabled",
            "colorError": "MuiSvgIcon-colorError",
            "colorPrimary": "MuiSvgIcon-colorPrimary",
            "colorSecondary": "MuiSvgIcon-colorSecondary",
            "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
            "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
            "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
            "root": "MuiSvgIcon-root",
          }
        }
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M2 20h20v-4H2v4zm2-3h2v2H4v-2zM2 4v4h20V4H2zm4 3H4V5h2v2zm-4 7h20v-4H2v4zm2-3h2v2H4v-2z"
          />
        </svg>
      </ForwardRef(SvgIcon)>
    </WithStyles(ForwardRef(SvgIcon))>
  </ForwardRef>
</OptionIconGenerator>
`;

exports[`Search renders without crashing 1`] = `
<OptionIconGenerator
  origin="user"
>
  <ForwardRef>
    <WithStyles(ForwardRef(SvgIcon))>
      <ForwardRef(SvgIcon)
        classes={
          Object {
            "colorAction": "MuiSvgIcon-colorAction",
            "colorDisabled": "MuiSvgIcon-colorDisabled",
            "colorError": "MuiSvgIcon-colorError",
            "colorPrimary": "MuiSvgIcon-colorPrimary",
            "colorSecondary": "MuiSvgIcon-colorSecondary",
            "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
            "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
            "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
            "root": "MuiSvgIcon-root",
          }
        }
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </ForwardRef(SvgIcon)>
    </WithStyles(ForwardRef(SvgIcon))>
  </ForwardRef>
</OptionIconGenerator>
`;

exports[`Search renders without crashing 2`] = `
<OptionIconGenerator
  origin="repo"
>
  <ForwardRef>
    <WithStyles(ForwardRef(SvgIcon))>
      <ForwardRef(SvgIcon)
        classes={
          Object {
            "colorAction": "MuiSvgIcon-colorAction",
            "colorDisabled": "MuiSvgIcon-colorDisabled",
            "colorError": "MuiSvgIcon-colorError",
            "colorPrimary": "MuiSvgIcon-colorPrimary",
            "colorSecondary": "MuiSvgIcon-colorSecondary",
            "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
            "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
            "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
            "root": "MuiSvgIcon-root",
          }
        }
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M2 20h20v-4H2v4zm2-3h2v2H4v-2zM2 4v4h20V4H2zm4 3H4V5h2v2zm-4 7h20v-4H2v4zm2-3h2v2H4v-2z"
          />
        </svg>
      </ForwardRef(SvgIcon)>
    </WithStyles(ForwardRef(SvgIcon))>
  </ForwardRef>
</OptionIconGenerator>
`;

exports[`user renders without crashing 1`] = `
<OptionIconGenerator
  origin="user"
>
  <ForwardRef>
    <WithStyles(ForwardRef(SvgIcon))>
      <ForwardRef(SvgIcon)
        classes={
          Object {
            "colorAction": "MuiSvgIcon-colorAction",
            "colorDisabled": "MuiSvgIcon-colorDisabled",
            "colorError": "MuiSvgIcon-colorError",
            "colorPrimary": "MuiSvgIcon-colorPrimary",
            "colorSecondary": "MuiSvgIcon-colorSecondary",
            "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
            "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
            "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
            "root": "MuiSvgIcon-root",
          }
        }
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </ForwardRef(SvgIcon)>
    </WithStyles(ForwardRef(SvgIcon))>
  </ForwardRef>
</OptionIconGenerator>
`;

exports[`user renders without crashing 2`] = `
<OptionIconGenerator
  origin="repo"
>
  <ForwardRef>
    <WithStyles(ForwardRef(SvgIcon))>
      <ForwardRef(SvgIcon)
        classes={
          Object {
            "colorAction": "MuiSvgIcon-colorAction",
            "colorDisabled": "MuiSvgIcon-colorDisabled",
            "colorError": "MuiSvgIcon-colorError",
            "colorPrimary": "MuiSvgIcon-colorPrimary",
            "colorSecondary": "MuiSvgIcon-colorSecondary",
            "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
            "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
            "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
            "root": "MuiSvgIcon-root",
          }
        }
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M2 20h20v-4H2v4zm2-3h2v2H4v-2zM2 4v4h20V4H2zm4 3H4V5h2v2zm-4 7h20v-4H2v4zm2-3h2v2H4v-2z"
          />
        </svg>
      </ForwardRef(SvgIcon)>
    </WithStyles(ForwardRef(SvgIcon))>
  </ForwardRef>
</OptionIconGenerator>
`;
